packages/router/__tests__/router-test.ts

```diff
@@ -11681,7 +11700,7 @@ describe("a router", () => {
       });
     });
 
-    it("resolves lazy route modules on submission navigation when navigating away before lazy promise resolves", async () => {
+    it("cancels lazy route modules on submission navigation when navigating away before lazy promise resolves", async () => {
```

Comment:
These tests made me realize we had some bugs in the implementation!  Not only do we need to not add to the router - we also want to short circuit the originating flow (`handleLoaders`, `handleAction`, etc.) and any potential attempt at calling loaders or actions.  Otherwise we could end up in a state where it thinks we don't have loaders or actions to run and "succeeds".